### PR TITLE
Add asian-southeast1 to RTDB CLI

### LINKED ISSUE #3455: Firebase init error with realtime database location asia-southeast1
I cannot init firebase project with firebase cli tool when checked with database in asia-southeast1

**firebase-tools:** 9.12.1

**Platform:** macOS

### [REQUIRED] Test case
Please refer to the following reproduce part

### [REQUIRED] Steps to reproduce
1. Create a firebase project from firebase web console
2. Go to database tab and enable realtime database
3. Select asia-southeast1 from the dropdown menu
4. Use firebase cli to init the project
5. Check database and finish init

### [REQUIRED] Expected behavior

I should be able to init the project with firebase-tools of the database
```

### [REQUIRED] Actual behavior
Error Message from firebase cli:

[2021-06-06T05:32:37.633Z] >>> HTTP REQUEST GET https://firebasedatabase.googleapis.com/v1beta/projects/gt-item-shop/locations/-/instances/gt-item-shop-default-rtdb

[2021-06-06T05:32:38.358Z] <<< HTTP RESPONSE 200 {"content-type":"application/json; charset=UTF-8","vary":"X-Origin, Referer, Origin,Accept-Encoding","date":"Sun, 06 Jun 2021 05:32:38 GMT","server":"ESF","cache-control":"private","x-xss-protection":"0","x-frame-options":"SAMEORIGIN","x-content-type-options":"nosniff","alt-svc":"h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\"","accept-ranges":"none","transfer-encoding":"chunked"}
[2021-06-06T05:32:38.359Z] Unexpected location value: asia-southeast1. Only us-central1, and europe-west1 locations are supported
[2021-06-06T05:32:38.360Z] FirebaseError: Unexpected location value: asia-southeast1. Only us-central1, and europe-west1 locations are supported
    at parseDatabaseLocation (/Users/jcyh/.nvm/versions/node/v13.11.0/lib/node_modules/firebase-tools/lib/management/database.js:132:19)
    at convertDatabaseInstance (/Users/jcyh/.nvm/versions/node/v13.11.0/lib/node_modules/firebase-tools/lib/management/database.js:173:19)
    at Object.getDatabaseInstanceDetails (/Users/jcyh/.nvm/versions/node/v13.11.0/lib/node_modules/firebase-tools/lib/management/database.js:44:16)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Object.doSetup [as database] (/Users/jcyh/.nvm/versions/node/v13.11.0/lib/node_modules/firebase-tools/lib/init/features/database.js:70:27)
    at async init (/Users/jcyh/.nvm/versions/node/v13.11.0/lib/node_modules/firebase-tools/lib/init/index.js:19:9)
```

### PR DESCRIPTION
**Pull request by fredzqm**
### Description

Fixes #3455

### Scenarios Tested

Create, read, edit, update rules for RTDB database in Singapore.

### Sample Commands

`firebase database:instances:create fredzqm-staging-apse1-2 --location asia-southeast1`
`firebase --instance fredzqm-staging-apse1 /`
`firebase deploy --only database`

### CODE REVIEW

**Comment by fredzqm** [comment]
Any idea why CI is failing. Did I forget to update in some places?

File: src/management/database.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export enum DatabaseInstanceState {
 export enum DatabaseLocation {
   US_CENTRAL1 = "us-central1",
   EUROPE_WEST1 = "europe-west1",
+  ASIAN_SOUTHEAST1 = "asia-southeast1",
```
Comment (bkendall) [inline src/management/database.ts:32]: The enum should match the name

```suggestion
  ASIA_SOUTHEAST1 = "asia-southeast1",
```

File: src/management/database.ts [hunk 2]
```diff
@@ -206,15 +207,17 @@ export function parseDatabaseLocation(
...
+    case "europe-west1":
+      return DatabaseLocation.EUROPE_WEST1;
+    case "asia-southeast1":
+      return DatabaseLocation.ASIAN_SOUTHEAST1;
     case "":
       return defaultLocation;
     default:
       throw new FirebaseError(
-        `Unexpected location value: ${location}. Only us-central1, and europe-west1 locations are supported`
+        `Unexpected location value: ${location}. Only us-central1, europe-west1 and asia-southeast1 locations are supported`
```
Comment (bkendall) [inline src/management/database.ts:220]: I like oxford commas?

```suggestion
        `Unexpected location value: ${location}. Only us-central1, europe-west1, and asia-southeast1 locations are supported`
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1 +1,2 @@
 - Add Warsaw (europe-central2) Cloud Function Location to Firebase Extension template.
+- Add Singapore (asia-southeast1) to Firebase Database location.
```
Comment (bkendall) [inline CHANGELOG.md:2]:
```suggestion
- Add Singapore (asia-southeast1) as a valid Firebase Realtime Database location.
```

**Review by bkendall — CHANGES_REQUESTED** [review]

File: src/management/database.ts (hunk 1 above)
Comment (fredzqm) [reply src/management/database.ts]: Oh, sorry, meant to keep them the same.

**Review by bkendall — APPROVED** [review]
LGTM, but take a look at the tests please :)

**Comment by fredzqm** [comment]
@bkendall I am suspecting the last commit https://github.com/firebase/firebase-tools/pull/3454 broken the CI build.
After I revert it on top of the this PR, all tests passes now: https://github.com/firebase/firebase-tools/pull/3461

**Comment by inlined** [comment]
Fix for the build break incoming.

**Review by samtstern — APPROVED** [review]

**Pull request merged**